### ISSUE TITLE
Cannot read property 'results_none_found' of undefined

### ISSUE THREAD

**Issue by ricardocasares** [leocaseiro/angular-chosen#177]
I had to update to 1.2.0 because I was getting the errors fixed in PR #164 
Since I had `~1.2.0` in my bower.json, whenI got the 1.2.1 update now I'm seeing this error in the console (Only when I'm in device emulation mode or in a real device). For now I'm fixing the version in my bower file.
```
TypeError: Cannot read property 'results_none_found' of undefined
    at y (http://localhost:3001/bower_components/angular-chosen-localytics/dist/angular-chosen.min.js:7:1833)
    at http://localhost:3001/bower_components/angular-chosen-localytics/dist/angular-chosen.min.js:7:2291
    at http://localhost:3001/bower_components/angular/angular.js:18744:31
    at completeOutstandingRequest (http://localhost:3001/bower_components/angular/angular.js:5804:10)
    at http://localhost:3001/bower_components/angular/angular.js:6081:7 undefined
```

**Comment by leocaseiro** [comment]
Hi @ricardocasares thank you for your report.

Could you please tell me more about your scenario. I'd like to emulate this error. So I could work on a better fix.

Is there a chance to post on a plunkr for me?

You can from from one of the [examples](leocaseiro.github.io/angular-chosen/#examples) that is more similar to your scenario.

Thanks

**Comment by leocaseiro** [comment]
Hi @ricardocasares I'm trying to emulate it still, when you mention device, you meant Mobile, like Safari iOS or Chrome Android? Thanks

**Comment by leocaseiro** [comment]
@ricardocasares and @greaterweb,
Would the PR #64 fix your issue? Should I accept that?

**Comment by ricardocasares** [comment]
Sorry @leocaseiro I haven't had time to create a fiddle to reproduce this, but I don't think #64 relates to this problem. I'll try to put an example together ASAP.

**Comment by ricardocasares** [comment]
It probably it's related more to #69, because it only happens when having mobile user agent enabled inside of chrome debugger or when trying from a real device.

**Comment by leocaseiro** [comment]
real device like a mobile?

**Comment by ricardocasares** [comment]
@leocaseiro yes, I mean a mobile device, not only emulated but also.
[This is the first](http://embed.plnkr.co/1U8KCH/) of the examples in the examples section in http://leocaseiro.github.io/angular-chosen/#examples you can just open that in embeded mode in plunkr, open DevTools and emulate any device.

![screen shot 2016-03-22 at 10 31 14 am](https://cloud.githubusercontent.com/assets/84963/13947357/570f5192-f01a-11e5-9912-cb74ddb6abc1.png)

**Comment by ricardocasares** [comment]
w

**Comment by ricardocasares** [comment]
I found one of the problems we were having on our setup for chosen, I think this can be closed.

**Comment by douglasward** [comment]
Hi @leocaseiro,

First of all, thanks for taking on this project and keeping it up to date! 

I believe this issue is still open, however. When viewing a chosen select box on a mobile device or an emulator (such as the mobile emulator in chrome) with ng-options referencing an empty array, I get the following error:

```
TypeError: Cannot read property 'results_none_found' of undefined
````

I have made a simple plnkr that reproduces the error. You just need to view it either in chrome using the iPhone emulator for example and refresh:
https://embed.plnkr.co/VPnse5TShEy9zfkCyu9r/

Any ideas?

Many thanks in advance.

**Comment by leocaseiro** [comment]
Hi @douglasward, thanks for reporting that.

I believe a simple ng-if would solve your issue. I'm not aware of a use case that you'd have an empty ng-options.

If you build like the example, would solve your issue?

`<select ng-if="ng-options.lenght > 0" chosen>...</select>`

**Comment by douglasward** [comment]
Using an ng-if would not trigger the error anymore, yes, but it is not the best ui option, and why then is there a `no-results-text` option...? 

The use case for us is the following: 
We have 2 select boxes, one of museums and one of exhibitions. The exhibitions select box is filtered by the museum one, so if one museum is selected that does not have any exhibitions linked to it, the exhibitions select box would be empty.

I noticed that on mobile devices, the placeholder and no-results text is not shown at all. Could this have something to do with this error?

**Comment by leocaseiro** [comment]
Hi @douglasward, agreed.

However, Chosen is disabled for mobile devices. I actually just [released v1.4.2 with a fix](https://github.com/leocaseiro/angular-chosen/commit/e40387ff45b19fce4ca4a1f5c06d0bee5400348f) for that, making sure we use native elements for mobile as recommended from [Chosen Team](https://github.com/harvesthq/chosen/pull/1388).

I believe we no longer will have this issue for mobiles.

Please let me know if I can help with more.

**Comment by douglasward** [comment]
Hi @leocaseiro, excellent! This solves it very nicely. Thank you very much.
